vtex-apps/product-list-context
# Reset previously sent events on ProductListContext upon list name changes

#### What problem is this solving?

The `ProductListContext` would not be reset across different product lists, and that would cause the `productImpression` events to be sent with inconsistent data, or not be sent at all.

This issue was reported as a KI, and more details can be found [in the Jira ticket](https://vtex-dev.atlassian.net/browse/IOSF-856).

#### How to test it?

1. Go to this [Workspace](https://victormiranda--easycl.myvtex.com/).
2. Scroll down a bit until you see a shelf of products.
3. Open your browser's console and print out the value of the `pixelManagerEvents` variable. You should see the following events listed there:

![Screen Shot 2022-03-10 at 18 48 35](https://user-images.githubusercontent.com/27777263/157760321-e60a7197-d3cd-4d3f-b98b-62e239cc5188.png)

Notice how the `productImpressions` event has indexes starting from 1.

4. Now navigate to any of the PLP pages from home.
5. Repeat step 3. You should see new `productImpression` events, starting from index 1 again.

#### Related to / Depends on

[Jira ticket](https://vtex-dev.atlassian.net/browse/IOSF-856)
